### PR TITLE
refactor(simple-agg): use fe generated table catalog for simple agg + some code-reuse refactor

### PR DESCRIPTION
**Pull request by BowenXiao1999**
…nd some simple refactor

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
* Add a method in mod.rs `generate_state_table_from_proto`. Currently reused by global agg and simple agg. 
* Add a method in test_util.rs `new_boxed_simple_agg_executor`. Currently used by tests in simple agg.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/stream/src/executor/global_simple_agg.rs [hunk 1]
```diff
@@ -92,27 +91,14 @@ impl<S: StateStore> SimpleAggExecutor<S> {
...
         key_indices: Vec<usize>,
+        state_tables: Vec<StateTable<S>>,
     ) -> Result<Self> {
         let input_info = input.info();
         let schema = generate_agg_schema(input.as_ref(), &agg_calls, None);
 
-        // Create state tables for each agg call.
-        let mut state_tables = Vec::with_capacity(agg_calls.len());
-        for (agg_call, ks) in agg_calls.iter().zip_eq(&keyspace) {
-            state_tables.push(generate_state_table(
```
Comment (BugenZhao) [inline src/stream/src/executor/global_simple_agg.rs:106]: Can we remove `generate_state_table` now?

Comment (BowenXiao1999) [reply src/stream/src/executor/global_simple_agg.rs:106]: It's now only used in unit test. I should add a test only tag

**Review by st1page — APPROVED** [review]
LGTM

File: src/stream/src/executor/global_simple_agg.rs (hunk 1 above)
Comment (BowenXiao1999) [reply src/stream/src/executor/global_simple_agg.rs:106]: Emmm I do not find one. Moved them all to test_utils.rs

**Pull request merged**